src/transformStateWithClones.js

```diff
@@ -7,7 +7,47 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+
+  let curentState = { ...state };
+
+  for (const action of actions) {
+    const { type = null, extraData = {}, keysToRemove = [] } = action;
+
+    switch (type) {
+      case 'addProperties':
+        addProperties(curentState, extraData);
+        break;
+      case 'removeProperties':
+        removeProperties(curentState, keysToRemove);
+        break;
+      case 'clear':
+        clear(curentState);
+        break;
+    }
```

Comment:
A default case should be added in the switch statement for error handling. This ensures that any unexpected action types are caught and handled appropriately.